src/plugins/dashboard/public/application/legacy_app.js

```diff
@@ -171,7 +170,18 @@ export function initDashboardApp(app, deps) {
             history.push(deps.addBasePath(viewUrl));
           };
           $scope.delete = (dashboards) => {
-            return service.delete(dashboards.map((d) => d.id));
+            const ids = dashboards.map((d) => ({ id: d.id, appId: d.appId }));
+            return Promise.all(
+              ids.map(({ id, appId }) => {
+                return deps.savedObjectsClient.delete(appId, id);
+              })
+            ).catch((error) => {
+              deps.toastNotifications.addError(error, {
+                title: i18n.translate('dashboard.dashboardListingDeleteErrorTitle', {
+                  defaultMessage: 'Error deleting dashboard',
+                }),
+              });
+            });
```

Comment:
If I understand this correctly, we fire off a bunch of `delete` requests in parallel and if one fails we show an error that there was a problem deleting the "dashboard" but in reality, we have deleted some stuff. This doesn't sound like good UX.

Reply:
I think we want to go with consistent UX (so that the same component on `dashboard` and `visualize` behaves the same) first, and then we can improve the existing UX in a follow-up.

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3877